### ISSUE TITLE
Improve dashboard dockerization

### ISSUE THREAD

**Issue by anupcowkur** [measure-sh/measure#1058]
- Remove multi stage docker file
- Remove bind mount
- Write a develop specification that sync target files on rebuild 
- Update contributing docs to remove the step where `npm install` is required to be run in `frontend/dashboard`

**Comment by detj** [comment]
Investigated this problem deeply. I was able to switch to a single stage docker build along with docker compose's develop and watch, target and path config and make HMR work as expected, ultimately, making these changes is not worth the effort.

That's because, even though the setup worked without a local `node_modules` directory, the imports in VSCode (and other text editors) fail because these text editors expect presence of a local `node_modules` in the project path. There is no straightforward way to make Language Server(s) work inside these text editors without a local `node_modules` directory.

Long story short, contributors will need [node](https://nodejs.org) available on their system path and they would need to run `npm install` command before starting the containers as earlier.

To make the process easier, I have updated the [CONTRIBUTING](../tree/main/docs/CONTRIBUTING.md) and [Self Host Guide](../tree/main/docs/hosting/README.md) to reflect this.
